### PR TITLE
Shanyin - Stepper Component

### PR DESCRIPTION
**Pull request by LIUSHANYIN**
a simple version stepper

### CODE REVIEW

**Comment by LIUSHANYIN** [comment]
shanyin/stepper

File: src/components/Stepper/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Stepper from '.';
+import '../../styles/stories.scss';
+
+export default {
+  title: 'Stepper',
+  component: Stepper
+};
+
+export const basicStepper = () => (
```
Comment (mxdi9i7) [inline src/components/Stepper/index.stories.tsx:10]: use pascal case for stories

File: src/components/Stepper/index.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+
+import classnames from '../../utils/classNames';
+import './index.scss';
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:4]: need a line between line 3 and 4

File: src/components/Stepper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  disabled?: boolean;
+  disableInput?: boolean;
+  min?: number | any;
+  max?: number | any;
+  step?: number | any;
+  longPress?: boolean;
+  plus?: boolean;
+  minus?: boolean;
+  size?: number;
+}
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:20]: need a line here

File: src/components/Stepper/index.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+
+import classnames from '../../utils/classNames';
+import './index.scss';
+
+const baseClass = 'vant-stepper';
+
+export interface Props {
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:8]: IProps

File: src/components/Stepper/index.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  min,
+  max,
+  disableInput,
+  size,
+  theme
+}: Props) {
+  const [value, setValue] = useState(1);
+  const [minus, setMinus] = useState(false);
+  const [plus, setPlus] = useState(false);
+  const [disInput, setDisInput] = useState(false);
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:33]: what is disInput? use a better name

File: src/components/Stepper/index.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+  disabled,
+  step,
+  min,
+  max,
+  disableInput,
+  size,
+  theme
+}: Props) {
+  const [value, setValue] = useState(1);
+  const [minus, setMinus] = useState(false);
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:31]: handleDecrement

File: src/components/Stepper/index.tsx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+  const [plus, setPlus] = useState(false);
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:32]: handleIncrement

File: src/components/Stepper/index.tsx [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+  const [disInput, setDisInput] = useState(false);
+  const plusBtProps = {
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:34]: name doesn't make sense

File: src/components/Stepper/index.tsx [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    className: classnames(baseClass, [{ disabled }, { theme }]),
+    style: {}
+  };
+  const minusBtProps = {
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:38]: name doesn't make sense

File: src/components/Stepper/index.tsx [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    className: classnames(baseClass, [{ disabled }, { theme }]),
+    style: {}
+  };
+  const inputProps = {
+    className: classnames(baseClass, [{}]),
+    style: {}
+  };
+
+  const handlePlus = () => {
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:47]: handleIncrement

File: src/components/Stepper/index.tsx [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+    const limit = max - 1 - value;
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:48]: what is that 1? use a constant name

File: src/components/Stepper/index.tsx [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+    if (step) {
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:49]: the logic here is too complicated, can you simplify?

File: src/components/Stepper/index.tsx [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+      setValue(value + step);
+    } else if (max && limit < 0) {
+      Object.assign(plusBtProps, { disabled });
+      setPlus(true);
+    } else {
+      setValue(value + 1);
+      setMinus(true);
+    }
+  };
+  const handleMinus = () => {
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:59]: simplify logic here

File: src/components/Stepper/index.tsx [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+    Object.assign(minusBtProps, { disabled });
+    Object.assign(plusBtProps, { disabled });
+    Object.assign(inputProps, { disabled });
+  }
+
+  if (size) {
+    Object.assign(minusBtProps, {
+      style: {
+        ...minusBtProps.style,
+        width: `${size}px`,
```
Comment (mxdi9i7) [inline src/components/Stepper/index.tsx:134]: you are using this so many times, can you abstract width and height?

**Pull request merged**